prog_test.go

```diff
@@ -90,6 +91,56 @@ func TestProgramRun(t *testing.T) {
 	}
 }
 
+func TestProgramRunWithOptions(t *testing.T) {
+	testutils.SkipOnOldKernel(t, "4.8", "XDP program")
+
+	ins := asm.Instructions{
+		// Return XDP_ABORTED
+		asm.LoadImm(asm.R0, 0, asm.DWord),
+		asm.Return(),
+	}
+
+	t.Log(ins)
+
+	prog, err := NewProgram(&ProgramSpec{
+		Name:         "test",
+		Type:         XDP,
+		Instructions: ins,
+		License:      "MIT",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer prog.Close()
+
+	buf := make([]byte, 14)
+	xdp := sys.XdpMd{
+		Data:    0,
+		DataEnd: 14,
+	}
+	xdpOut := sys.XdpMd{}
+	ctx := (*(*[unsafe.Sizeof(xdp)]byte)(unsafe.Pointer(&xdp)))[:]
```

Comment:
See my comment on `binary.Write` above, usage of `unsafe` should not be necessary / encouraged.